gprofiler/cloud_metadata.py

```diff
@@ -0,0 +1,161 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+import requests
+from requests import Response
+
+from gprofiler.exceptions import BadResponseCode
+from gprofiler.log import get_logger_adapter
+
+METADATA_REQUEST_TIMEOUT = 5
+
+logger = get_logger_adapter(__name__)
+
+
+@dataclass
+class InstanceMetadataBase:
+    provider: str
+
+
+@dataclass()
+class AwsInstanceMetadata(InstanceMetadataBase):
```

Comment:
Document these classes. Where did you get their structure from (web link to relevant AWS/GCP/whatever page)